### PR TITLE
Recipe Changes for AL2022 Neuron Support

### PR DESCRIPTION
**Pull request by Realmonia**
### Summary

This PR adds a new build target - `al2022neu`, and adds `scripts/enable-ecs-agent-inferentia-support.sh` to AL2022 recipes.

### Implementation details

* Add new target to Makefile
* Modify `al2022.pkr.hcl` build sources to include `al2022neu`, as well as two provisioners to
  * run `scripts/enable-ecs-agent-inferentia-support.sh` to install neuron packages on AL2022 base instance and enable `INF` on instance.
  * install necessary neuron packages that is not in AL2022 Repo yet in `install-neuron-packages.sh` (missed packages can be found in `scripts/al2022/install-neuron-packages.sh`). Downloaded AL2 version of those rpms and put them under `amazon-ecs-ami/additional-packages/al2022`
* Create `al2022neu.pkr.hcl` and `scripts/al2022/install-neuron-packages.sh`
* Modify `scripts/enable-ecs-agent-inferentia-support.sh` to also run when `AMI_TYPE` is `al2022neu` 

### Testing

New tests cover the changes:  No

Manual Tests:
Ran `REGION=us-west-2 make al2022neu` and successfully created an AMI (ami-056e2478617a095b7 in personal test account) with name `unofficial-amzn2022-ami-ecs-neu-hvm-2022.0.20220520-x86_64-ebs`

Launched an INF instance with this AMI and ran a busybox task on this instance.

### Description for the changelog

Recipe changes for AL2022 Neuron Support

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/al2022/install-neuron-packages.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (sparrc) [inline scripts/al2022/install-neuron-packages.sh:1]: Can we just reuse the existing install-additional-packages script for this?

File: al2022neu.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+locals {
+  ami_name_al2022neu = "${var.ami_name_prefix_al2022}-neu-hvm-2022.0.${var.ami_version}-x86_64-ebs"
```
Comment (sparrc) [inline al2022neu.pkr.hcl:2]: Not sure but is there any particular reason to keep the platform type 3 letters? Wondering if maybe we should call it `neuron-hvm-2022...` for the external name

Comment (Realmonia) [reply al2022neu.pkr.hcl:2]: I don't have specific reason for this and I prefer full name too 😂 Just following the convention

File: scripts/al2022/install-neuron-packages.sh (hunk 1 above)
Comment (Realmonia) [reply scripts/al2022/install-neuron-packages.sh:1]: I am getting errors like
```
==> amazon-ebs.al2022neu: + sudo yum localinstall -y '/tmp/additional-packages/*.x86_64.rpm'
    amazon-ebs.al2022neu: Last metadata expiration check: 0:00:31 ago on Thu 26 May 2022 09:19:27 PM UTC.
==> amazon-ebs.al2022neu: Can not load RPM file: /tmp/additional-packages/*.x86_64.rpm.
==> amazon-ebs.al2022neu: Could not open: /tmp/additional-packages/*.x86_64.rpm
```
When I have noarch rpms under this directory. I am not really sure why it happened though...

Comment (Realmonia) [reply scripts/al2022/install-neuron-packages.sh:1]: yep I will take a look

**Review by sparrc — APPROVED** [review]

**Review by singholt — APPROVED** [review]

**Comment by Realmonia** [comment]
Update on PR testing: used Amazon Linux team's beta build of dkms and oci package to build AL2022 neuron AMI and it is working. Next steps before merging into main branch is
1. wait for AL packages to be pushed to repo
2. test with an Trn instance

**Pull request merged**